src/NewPost.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from "react";
+
+export class NewPost extends Component {
+  state = {
+    title: "",
+    author: "",
+    createdOn: "",
+    body: "",
+    tags: []
+  };
+
+  handleChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+  };
+
+  render() {
+    return (
+      <form>
+        <input
+          type="text"
+          name="title"
+          placeholder="blog title"
+          value={this.state.title}
+          onChange={e => this.handleChange(e)}
+        />
+        <br />
+        <input
+          type="text"
+          name="author"
+          placeholder="author"
+          value={this.state.author}
+          onChange={e => this.handleChange(e)}
+        />
+        <br />
+        <input
+          type="date"
+          name="createdOn"
+          placeholder="createdOn"
+          value={this.state.createdOn}
+          onChange={e => this.handleChange(e)}
+        />
+        <br />
+        <input
+          type="textarea"
+          name="body"
+          placeholder="body"
+          value={this.state.body}
+          onChange={e => this.handleChange(e)}
+        />
+        <br />
+        <input
+          type="text"
+          placeholder="tags"
+          value={this.state.tags}
+          onChange={e => this.setState({ tags: e.target.value })}
```

Comment:
`tags` is supposed to be an array of strings, but what do you see as its value if you enter some tags, submit the form, and log the form state in the `handleSubmit` function? `tags` will need its own change handler to convert the raw value from the input into an array. that means we need to decide a rule for the `tags` input to follow so we can do that conversion. i suggest requiring tags be entered with commas to separate them. then your tag change handler can look for commas and convert the string value to an array of strings